docs/guide/snaps-rpc-api.md

```diff
@@ -447,6 +447,166 @@ if (result === true) {
 }
 ```
 
+### `snap_dialog`
+
+::: warning Only Callable By
+
+- Snaps
+  :::
+
+#### Description
+
+Calling this method causes a dialog to be displayed in the MetaMask UI.
+There are three types of dialogs: Alert, Confirmation, and Prompt.
+Each of these dialog types has different parameters and return types, detailed below.
+
+#### The `Component` type
+
+The `Component` type, used as the `content` property for dialogs, is [defined in source code](https://github.com/MetaMask/snaps-monorepo/blob/main/packages/snaps-ui/src/nodes.ts#L171) as:
+
+```typescript
+type Component = Infer<typeof ComponentStruct>;
+```
```

Comment:
This explanation won't really make sense to developers not familiar with our source code. Once #616 is merged, we should link to the Snaps UI explanation instead.

Reply:
I'm not sure that the Custom UI documentation explains what `Component` as a type is. I feel like its explanation is internal and is bound to change.